AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/SQLite/StorageEngineAdapter+UntypedModel.swift

```diff
@@ -33,33 +35,51 @@ extension SQLiteStorageEngineAdapter {
             let shouldUpdate = try exists(modelSchema,
                                           withIdentifier: untypedModel.identifier(schema: modelSchema))
 
-            // TODO serialize result and create a new instance of the model
-            // (some columns might be auto-generated after DB insert/update)
             if shouldUpdate {
                 let statement = UpdateStatement(model: untypedModel, modelSchema: modelSchema)
                 _ = try connection.prepare(statement.stringValue).run(statement.variables)
             } else {
                 let statement = InsertStatement(model: untypedModel, modelSchema: modelSchema)
                 _ = try connection.prepare(statement.stringValue).run(statement.variables)
             }
-
-            completion(.success(untypedModel))
+            
+            query(modelSchema: modelSchema,
+                  predicate: untypedModel.identifier(schema: modelSchema).predicate,
+                  eagerLoad: eagerLoad) {
```

Comment:
This method `save` used to return a model from the input, which is now inaccurate in the lazy loading cases. The input model `untypedModel` may be a model that was from the response of an Amplify.API mutation call. Imagine returning this model back to the caller in a MutationEvent on the DataStore Observe API. This will mean that the lazy loading functionality will delegate to the AppSyncModelProvider and AppSyncListProvider, calling into API directly. To avoid this, we do a query to get the latest data from the local store and return that data to the caller (model -> MutationEvent -> emit to Observe API)